includes/controllers/pages/class-submit-listing.php

```diff
@@ -1151,8 +1151,8 @@ private function listing_images() {
 				$updated_meta = wpbdp_get_var( array( 'param' => 'images_meta' ), 'post' );
 				$updated_meta = ! empty( $updated_meta[ $img_id ] ) ? (array) $updated_meta[ $img_id ] : array();
 
-				$new_order = ! empty( $updated_meta['order'] ) ? intval( $updated_meta['order'] ) : $order;
-				update_post_meta( $img_id, '_wpbdp_image_weight', $thumbnail_id ? 0 : $new_order );
+				$new_order = ! empty( $updated_meta['order'] ) ? absint( $updated_meta['order'] ) : $order;
+				update_post_meta( $img_id, '_wpbdp_image_weight', $new_order );
```

Comment:
Does the thumbnail still get set correctly? If it's working, it looks like we don't need the $thumbnail_id lines above (1133-1139).

Reply:
Line `1133-1137` have been maintained to set the thumbnail on post. The other `else` has been removed. The thubmnail is set correctly in the first `if` statement

Reply:
Thanks!